python/ray/autoscaler/aws/node_provider.py

```diff
@@ -41,6 +42,8 @@ class AWSNodeProvider(NodeProvider):
     def __init__(self, provider_config, cluster_name):
         NodeProvider.__init__(self, provider_config, cluster_name)
         config = Config(retries={"max_attempts": BOTO_MAX_RETRIES})
+        self.cache_stopped_nodes = provider_config.get("cache_stopped_nodes",
+                                                       True)
```

Comment:
this changes default behavior right? shouldn't this default to False for now?

Reply:
Yep. I think this is what most people want, so let's try it out. It prints out a warning so it's not hard to figure out the change in behaviour.

Reply:
This also doesn't work for spot instances by default. 

```
╭─ ~/Research/riselab/sosp2019/scripts/cluster_cfg 
╰─ ray down ablation.yaml
This will destroy your cluster [y/N]: y
2019-09-22 15:36:33,323	INFO commands.py:111 -- teardown_cluster: Terminating 1 nodes...
2019-09-22 15:36:33,323	INFO node_provider.py:320 -- AWSNodeProvider: stopping nodes ['i-043519194fe3f6853']. To terminate nodes on stop, set 'cache_stopped_nodes: False' in the provider config.
2019-09-22 15:36:33,584	INFO log_timer.py:21 -- teardown_cluster: Termination done. [LogTimer=261ms]
Traceback (most recent call last):
  File "/Users/rliaw/miniconda3/bin/ray", line 10, in <module>
    sys.exit(main())
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/scripts/scripts.py", line 807, in main
    return cli()
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/scripts/scripts.py", line 516, in teardown
    teardown_cluster(cluster_config_file, yes, workers_only, cluster_name)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/autoscaler/commands.py", line 112, in teardown_cluster
    provider.terminate_nodes(A)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/ray/autoscaler/aws/node_provider.py", line 321, in terminate_nodes
    self.ec2.meta.client.stop_instances(InstanceIds=node_ids)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/botocore/client.py", line 357, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/rliaw/miniconda3/lib/python3.7/site-packages/botocore/client.py", line 661, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (UnsupportedOperation) when calling the StopInstances operation: The instance 'i-043519194fe3f6853' is a spot instance and may not be stopped.
```

With an unactionable error message.

Reply:
well actually, I guess it's on the very top...

Reply:
Fixed